diff --git a/dist/index.js b/dist/index.js
index ce558fffa1d6deaa53cd8edff398d24c3eee8ed3..1c88c1cc7c7fbc7b094c9de5ce768dc5b86b6e0f 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -69708,7 +69708,8 @@ __export(src_exports, {
   prepareCache: () => prepareCache,
   shouldServe: () => import_build_utils7.shouldServe,
   startDevServer: () => startDevServer,
-  version: () => version
+  version: () => version,
+  getRegExpFromMatchers: () => getRegExpFromMatchers
 });
 module.exports = __toCommonJS(src_exports);
 
@@ -70733,5 +70734,6 @@ var version = 3;
   prepareCache,
   shouldServe,
   startDevServer,
-  version
+  version,
+  getRegExpFromMatchers
 });
